ecommerce/extensions/api/v2/views/catalog.py

```
@@ -73,3 +73,21 @@ def preview(self, request):
                 logger.error('Unable to connect to Course Catalog service.')
                 return Response(status=status.HTTP_400_BAD_REQUEST)
         return Response(status=status.HTTP_400_BAD_REQUEST)
+
+    @action(is_for_list=True, methods=['get'])
+    def course_catalogs(self, request):
+        """
+        Returns response with all course catalogs in the format:
+        ["results": {"id": 1, "name": "Dummy Catalog"}]
+        """
+        try:
+            results = get_course_catalogs(site=request.site)
+        except (ConnectionError, SlumberBaseException, Timeout):
```

It looks like this except may not be hit because `get_course_catalogs` has a catch-all `except` handler that swallows all exceptions and returns `no_data`